### PR TITLE
ddl, transaction: DDL on temporary tables won't affect transactions

### PR DESCRIPTION
**Pull request by djshow832**
### What problem does this PR solve?

Problem Summary:
Schema change on a global temporary table may happen during a transaction that writes to the temporary table. Unlike normal tables, the transaction won’t overwrite other transactions, so it’s fine to commit. 

### What is changed and how it works?

What's Changed:
Do not put temporary tables into `SchemaChecker.relatedTableIDs`.

### Related changes

- N/A

### Check List 

Tests 

- Unit test

Side effects

- N/A

### Release note 

- DDL on temporary tables won't affect transactions.

### CODE REVIEW

File: domain/schema_validator.go [hunk 1]
```diff
@@ -234,8 +234,9 @@ func (s *schemaValidator) Check(txnTS uint64, schemaVer int64, relatedPhysicalTa
 
 	// Schema changed, result decided by whether related tables change.
 	if schemaVer < s.latestSchemaVer {
-		// The DDL relatedPhysicalTableIDs is empty.
-		if len(relatedPhysicalTableIDs) == 0 {
+		// When a transaction executes a DDL, relatedPhysicalTableIDs is nil.
+		// When a transaction only contains DML on temporary tables, relatedPhysicalTableIDs is [].
+		if relatedPhysicalTableIDs == nil {
```
Comment (tiancaiamao) [inline domain/schema_validator.go:239]: Why not just use `len(relatedPhysicalTableIDs) == 0`? It looks more intuitive.
It's a burden for the reader to distinguish `nil` and `[]` and search the corresponding context to understand the code.

**Review by tiancaiamao — COMMENTED** [review]
Rest looks good to me.

File: domain/schema_validator.go (hunk 1 above)
Comment (djshow832) [reply domain/schema_validator.go:239]: The reason is in the comment. 
The code block leads to failure, but I don't want it to fail where there's nothing to commit.
Any better ideas?
I also tried to remove SchemaChecker when there are no normal tables to commit, but some logic needs a SchemaChecker.

Comment (tiancaiamao) [reply domain/schema_validator.go:239]: This time I understand their difference.
`relatedPhysicalTableIDs == nil` => deny
`relatedPhysicalTableIDs == []` => accept

**Pull request merged**